test/metabase/query_processor_test/explicit_joins_test.clj

```diff
@@ -324,6 +326,80 @@
                      :order-by [[:asc $name]]
                      :limit    3})))))))))
 
+;; This is a very contrived test. We create two identical cards and join them both
+;; in a third card. This means that first two cards bring fields that differ only in
+;; their join aliases, but these "get lost" when the third card is joined to a table.
+;; Since we have multiple fields with the same ID, the IDs cannot be used to
+;; unambiguously refer to the underlying fields. The only way to reference them
+;; properly is by the name they have in the source metadata.
+(deftest ^:parallel join-against-multiple-card-copies-test
```

Comment:
If I'm understanding this correctly, I think something like `TODO: When this middleware is ported to MLv2, the aliases can be connected properly.` would be appropriate here.

Reply:
Would you really add this comment to this test? This test case will be relevant no matter what middlewares we have. I expect that only the way the query is built will have to change.